sklearn/feature_selection/_from_model.py

```diff
@@ -23,8 +23,10 @@ def _calculate_threshold(estimator, importances, threshold):
         # determine default from estimator
         est_name = estimator.__class__.__name__
         if (
-            hasattr(estimator, "penalty") and estimator.penalty == "l1"
-        ) or "Lasso" in est_name:
+            (hasattr(estimator, "penalty") and estimator.penalty == "l1")
+            or "Lasso" in est_name
+            or ("ElasticNet" in est_name and estimator.l1_ratio == 1)
```

Comment:
We likely need to special case `ElasticNetCV` as well. For `ElasticNetCV`, it stores the best `l1_ratio` as `l1_ratio_`.

By special casing both `ElasticNetCV` and `ElasticNet`, I prefer to be stricter:

```python
or (isinstance(estimator, ElasticNet) and estimator.l1_ratio == 1)
or (isinstance(estimator, ElasticNetCV) and estimator.l1_ratio_ == 1)
```

Reply:
* I found that there are also `LassoCV`. Is this covered by `"Lasso" in est_name`?
* About the stricter ElasticNetCV checks:
    * Does that means we only use threshold 1e-5 on ElasticNet which considered l1_ratio = 1 is the best after its cross-validation?
    * Should we also checks and tests `MultiTaskElasticNet`(CV) similar to `ElasticNet`(CV)?

Reply:
I'm okay with doing `hasattr(l1_ratio, ...)`. My biggest concern is if a third party estimator has a `l1_ratio` parameter that does not have the same semantics as `ElasticNet`.

I think you suggestion works as long as `l1_ratio_` is checked first. `ElasticNetCV.l1_ratio` accepts a list of floats, which would fail the `== 1` check.

Hmm, here another version is:

```python
"ElasticNet" in est_name and
hasattr(estimator, "l1_ratio_") and estimator.l1_ratio_ == 1 or
hasattr(estimator, "l1_ratio") and np.isscalar(estimator.l1_ratio) and estimator.l1_ratio == 1
```

This should cover all the `*ElasticNet*` classes. Checking for the name "ElasticNet" is not as strict as `isinstance`, but if "ElasticNet" is in the name then I think it's reasonable to assume `l1_ratio` has the correct semantics.

Reply:
Thanks for the review!
I agree with your consideration and this should cover all the `*ElasticNet*` classes.

But I think we might not need to guard against the check if `l1_ratio` is a list of floats.
Since from my understanding, we only use 1e-5 threshold for `ElasticNetCV` with `l1_ratio_ == 1`.

What's more, it seems that if  `estimator.l1_ratio` is a list of floats (e.g. `[1]`), both
`np.isscalar(estimator.l1_ratio) and estimator.l1_ratio == 1` and 
`estimator.l1_ratio == 1` will return `False`.
Therefore, I think it does not need `np.isscalar(estimator.l1_ratio)` check. 
```python
import numpy as np
a = [1.0, 0.9, 0.8]
condition_a = (np.isscalar(a) and a == 1)
condition_b = (a == 1)
print(condition_a, condition_b)
```